src/main/scala/org/apache/spark/shuffle/ucx/CommonUcxShuffleManager.scala

```diff
@@ -57,24 +68,33 @@ abstract class CommonUcxShuffleManager(val conf: SparkConf, isDriver: Boolean) e
    * Atomically starts UcxNode singleton - one for all shuffle threads.
    */
   def startUcxTransport(): Unit = if (ucxTransport == null) {
+    logInfo(s"LEO startUcxTransport")
     val blockManager = SparkEnv.get.blockManager.blockManagerId
     val transport = new UcxShuffleTransport(ucxShuffleConf, blockManager.executorId.toLong)
-    val address = transport.init()
-    ucxTransport = transport
+    transport.init()
+
     val rpcEnv = RpcEnv.create("ucx-rpc-env", blockManager.host, blockManager.port,
       conf, new SecurityManager(conf), clientMode = false)
-    executorEndpoint = new UcxExecutorRpcEndpoint(rpcEnv, ucxTransport, setupThread)
+    val driverEndpoint = RpcUtils.makeDriverRef(driverRpcName, conf, rpcEnv)
+
+    executorEndpoint = new UcxExecutorRpcEndpoint(rpcEnv, transport, setupThread,
+        ucxTransport, driverEndpoint, blockManager.executorId.toLong, setTransport)
     val endpoint = rpcEnv.setupEndpoint(
       s"ucx-shuffle-executor-${blockManager.executorId}",
       executorEndpoint)
-    val driverEndpoint = RpcUtils.makeDriverRef(driverRpcName, conf, rpcEnv)
+    logInfo(s"LEO startUcxTransport sending RPC IntroduceAllExecutors")
+
+    var sockAddress = new InetSocketAddress(DpuUtils.getLocalDpuAddress(), 1338)
```

Comment:
Consider declaring 1338 as a constant?

Reply:
It actually not constant. It's depends in the spark_service listening port.
IMO better approach will be to change cluster.ini file so it will include DPU IP address and port.

@gleon99 WDYT?